src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -9,94 +10,183 @@ internal struct FileStatus
     {
         private const int NanosecondsPerTick = 100;
 
+        // The last cached lstat information about the file
+        private Interop.Sys.FileStatus _fileCache;
+
+        // -1: if the file cache isn't initialized - Refresh should always change this value
+        //  0: if the file cache was initialized with no errors
+        // Positive number: the error code returned by the lstat call
+        private int _initializedFileCache;
+
         // The last cached stat information about the file
-        private Interop.Sys.FileStatus _fileStatus;
+        // Refresh only collects this if lstat determines the path is a symbolic link
+        private Interop.Sys.FileStatus _symlinkCache;
 
-        // -1 if _fileStatus isn't initialized, 0 if _fileStatus was initialized with no
-        // errors, or the errno error code.
-        private int _fileStatusInitialized;
+        // -1: if the symlink cache isn't initialized - Refresh only changes this value if lstat determines the path is a symbolic link
+        // 0: if the symlink cache was initialized with no errors
+        // Positive number: the error code returned by the stat call
+        private int _initializedSymlinkCache;
 
         // We track intent of creation to know whether or not we want to (1) create a
         // DirectoryInfo around this status struct or (2) actually are part of a DirectoryInfo.
-        internal bool InitiallyDirectory { get; private set; }
+        // Set to true during initialization when the DirectoryEntry's INodeType describes a directory
+        internal bool InitiallyDirectory { get; set; }
 
         // Is a directory as of the last refresh
-        internal bool _isDirectory;
+        // Its value can come from either the main path or the symbolic link path
+        private bool _isDirectory;
 
         // Exists as of the last refresh
         private bool _exists;
 
-        internal static void Initialize(
-            ref FileStatus status,
-            bool isDirectory)
+        private bool IsFileCacheInitialized    => _initializedFileCache == 0;
+        private bool IsSymlinkCacheInitialized => _initializedSymlinkCache == 0;
+
+        private bool AreCachesInitialized =>
```

Comment:
Since this is only used in one place, could you please move the condition there and remove this property?